RegionalTestApp/Program.cs

```diff
@@ -0,0 +1,107 @@
+            X509Certificate2 certificate = ReadCertificate("97D8C9DB3C84874D0363DCA540778461B2291780");
+
+            Console.WriteLine("=== Acquire token regional ===");
+            await AcquireTokenAsync(true, certificate).ConfigureAwait(false);
+            Console.WriteLine();
+
+            Console.WriteLine("=== Acquire token global ===");
+            await AcquireTokenAsync(false, certificate).ConfigureAwait(false);
+
+        }
+
+        private static void Log(LogLevel level, string message, bool containsPii)
+        {
+            if (containsPii)
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+            }
+            Console.WriteLine($"{level} {message}");
+            Console.ResetColor();
+        }
+
+        private static X509Certificate2 ReadCertificate(string thumbprint)
+        {
+            if (string.IsNullOrWhiteSpace(thumbprint))
+            {
+                throw new ArgumentException("certificateName should not be empty. Please set the CertificateName setting in the appsettings.json", "certificateName");
+            }
+            X509Certificate2 cert = null;
+
+            using (X509Store store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            {
+                store.Open(OpenFlags.ReadOnly);
+                X509Certificate2Collection certCollection = store.Certificates;
+
+                // Find unexpired certificates.
+                X509Certificate2Collection currentCerts = certCollection.Find(X509FindType.FindByTimeValid, DateTime.Now, false);
+
+                // From the collection of unexpired certificates, find the ones with the correct name.
+                X509Certificate2Collection signingCert = currentCerts.Find(X509FindType.FindByThumbprint, thumbprint, false);
+
+                // Return the first certificate in the collection, has the right name and is current.
+                cert = signingCert.OfType<X509Certificate2>().OrderByDescending(c => c.NotBefore).FirstOrDefault();
+            }
+            return cert;
+        }
+
+        private static async Task AcquireTokenAsync(bool withAzureRegion, X509Certificate2 certificate)
+        {
+            string[] scopes = new string[] { $"{clientId}/.default", };
+
+            var cca = ConfidentialClientApplicationBuilder.Create(clientId)
+                .WithAuthority("https://login.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47")
+                .WithCertificate(certificate)
+                .WithExperimentalFeatures(true)
+                .WithLogging(Log, LogLevel.Info, true)
+                .Build();
+
+            Console.WriteLine("CCA created");
+
+            var result = await cca.AcquireTokenForClient(scopes)
+            .WithAzureRegion(withAzureRegion)
+            .ExecuteAsync()
+            .ConfigureAwait(false);
+
+            Console.WriteLine("Access token:" + result.AccessToken);
```

Comment:
is there anything else in the result that would be interesting to make assure the test worked as expected? I don't remember the specific values in the auth result, but maybe you can print more info to the console.

Reply:
There are logs that print on the console and it contains more info. I have added what to check from logs in the steps. I was checking other tests and did not find anything else that might fit in here. Let me know if you have suggestions for something specific you think could be added to the logs. I might not be aware of it.